packages/client-mav/src/app/make-a-vis/data-view/shared/data.service.ts

```diff
@@ -62,24 +63,24 @@ export class DataService {
   getSelectableDataVariables(dataVariables: DataVariable[], graphicVariables: GraphicVariable[]) {
     const selectableDataVariable: Selectable = {
       color : [],
-      areaSize: []
+      areaSize: [],
+      axis: []
     };
     dataVariables.forEach((dv: DataVariable) => {
       graphicVariables.forEach((gv: GraphicVariable) => {
         if (gv.dataVariable.id === dv.id) {
-          if (gv.type === 'color') {
-          selectableDataVariable.color.push(dv);
-          }
-
-          if (gv.type === 'areaSize') {
-            selectableDataVariable.areaSize.push(dv);
+          switch (gv.type) {
```

Comment:
This should be simplified so you won't have to repeat the specific gv.types by doing this instead:
```javascript
if (selectableDataVariable.hasOwnProperty(gv.type)) {
    selectableDataVariable[gv.type].push(gv);
}
```

Reply:
I have made this change, but instead of `push(gv)` it should be `push(dv)`. Please review.